fix: updates to required JS projects

scissorsneedfoodtoo opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Related to https://github.com/freeCodeCamp/freeCodeCamp/pull/52370

<!-- Feel free to add any additional description of changes below this line -->

jdwilkin4 commented:
Hey @scissorsneedfoodtoo ! 

I will have more time to review this later on this week.
But one thing I have noticed in these projects is that we are using `innerHTML` for everything when we really should reserve that for when we are setting new HTML. If we are just setting text, then we should use another method like `textContent`.

scissorsneedfoodtoo commented:
Hey @jdwilkin4, thanks for taking a quick look at this and catching the issue with using `innerHTML` rather than `innerText`. That's something I missed while originally reviewing these projects.

I can go in and fix those, and apply the updated code to the Learn PR solutions.

jdwilkin4 reviewed (changes requested):
Looks good.
Left a couple of comments 👍

scissorsneedfoodtoo commented on apps/pokemon-search-app/index.html:
```diff
@@ -38,7 +30,7 @@ <h1>Pokémon Search App</h1>
             <span id="height"></span>
           </div>
-          <img src alt id="sprite" />
+          <div class="sprite-container"><img alt id="sprite" /></div>
```
Thanks for taking a look at this, @jdwilkin4, and for running the code though a validator. That's a great idea – I've just been using Prettier this whole time, not checking for actual validation errors.

We did have a default image before with a Pikachu sprite, but it was decided that we should remove that just to be extra safe.

We could go with a black box there, just a `div` with a background color and swap it out for an `img` element whenever appropriate. Or we could just add an `img` there whenever we get a response from the API proxy and remove it whenever there are errors.

I'll play around with it and push an update ASAP.

scissorsneedfoodtoo replied:
This should be fixed in the demo here, and in the solution in the Learn PR.

scissorsneedfoodtoo commented on apps/pokemon-search-app/public/script.js:
```diff
@@ -22,57 +22,52 @@ const getPokemon = async () => {
   // reset image src and alt
-  sprite.src = "";
-  sprite.alt = "";
+  sprite.src = '';
```
This should be fixed in the demo here, and in the solution in the Learn PR.

jdwilkin4 reviewed (approved):
Looks good 👍

naomi-lgbt reviewed (commented):
Couple of nits

jdwilkin4 commented on apps/palindrome-checker/public/script.js:
```diff
@@ -1,24 +1,25 @@
   }
 
-  if (hasSpecialCharactersOrDigits === true) {
+  if (hasSpecialCharactersOrDigits) {
     alert('Input should not include numbers and special characters');
     return;
```
I think @scissorsneedfoodtoo mentioned there was some sort of issue about having them strip the characters. 
There was a prior discussion thread in the old staff chat but I don't remember the exact reason why this was a problem.
So @scissorsneedfoodtoo should have the details on that.

scissorsneedfoodtoo replied:
The last messages in chat were about the demo project incorrectly flagging spaces as special characters. I can't remember exactly why we have this alert for special characters, or when it got added.

I'm in favor of updating the project so it's closer to the current palindrome checker, where special characters and spaces are stripped out before checking for a palindrome. It shouldn't be too difficult to update the solution here and add a couple of extra tests.

scissorsneedfoodtoo replied:
The Palindrome Check here and in https://github.com/freeCodeCamp/freeCodeCamp/pull/52370 have been updated so they're closer to the solution we have in the current JS certification where whitespace and special characters are stripped before checking for a palindrome.

scissorsneedfoodtoo commented on apps/roman-numeral-converter/index.html:
Done! It didn't take all that long to swap some of the colors around. I'll post screenshots of the latest Roman Numeral Converter styling below so they're easier for everyone to see.

scissorsneedfoodtoo commented:
Here's what the dark mode version of the Roman Numeral Converter project looks like:

![image](https://github.com/freeCodeCamp/demo-projects/assets/2051070/bc12587d-23ee-42dd-99e8-ab863a4d1ee7)

![image](https://github.com/freeCodeCamp/demo-projects/assets/2051070/f3128445-9ab3-4477-97ca-fb0105867340)

scissorsneedfoodtoo commented:
Added a logo and adjusted the styling of the Pokémon Search app slightly:

![image](https://github.com/freeCodeCamp/demo-projects/assets/2051070/93d7de13-9438-4c49-8be3-6d80cc9c9e0a)

naomi-lgbt reviewed (approved):
<img width="417" alt="image" src="https://github.com/freeCodeCamp/demo-projects/assets/63889819/ba492ffe-65ba-4c77-8a58-cc9edea97afe">

Okay, everything is looking good on my end. Let's go ahead and land this. Thanks for all your work on these!
